client/register.go

```diff
@@ -49,6 +49,7 @@ func runRegister(cmd *Command, args []string) {
 	k := NewKeysFile(daemonFolder + daemonToRegister)
 
 	var err error
+	var cmdName string
 	var ks []string
 	if *registerKey == "" {
```

Comment:
I think this check, and the one at the start of the method are still going to trip things up and force users to write `knox register -r -k foo` instead of `knox register -r`

Reply:
yeah, the `Overwrite` call is taking the whole list of keys, not just one: https://github.com/pinterest/knox/blob/80bb2bbb64c012e0d2e1064bfd508a75067015cb/client/daemon.go#L428-L443

So the docs are still correct to say `-r removes all existing registered keys.`. Therefore we shouldn't be looking expecting a `-k` arg.

Reply:
ack I misunderstood the usage here, will update.